packages/algob/src/lib/lsig.ts

```diff
@@ -17,3 +18,22 @@ export async function getLsig (name: string, scParams: Object, algodClient: algo
   const program = result.toBytes;
   return algosdk.makeLogicSig(program, scParams);
 }
+
+/**
+ * Description: this function decodes msig object from logic signature
+ * @param {string} msig : multisigned msig obj
+ * @returns {Object} : decoded msig (object with decoded public keys and their signatures)
+ */
+export async function decodeMsigObj (msig: string): Promise<Object> {
+  const parsedMsig = JSON.parse(msig).msig;
+
+  // decoding multisigned logic signature
+  for (const acc of parsedMsig.subsig) {
+    const decoded = decode.asBytes(acc.pk);
+    acc.pk = new Uint8Array(decoded.slice(0, 32)); // decode public key
+    if (acc.s) { // decode if addr is signed
+      acc.s = new Uint8Array(Buffer.from(acc.s, 'base64'));
```

Comment:
why here you are using `Buffer.from` and few lines earlier `decode.asBytes`?

Reply:
`line 32` decodes a base32 string (account addr) to bytes
`line 35` is decoding a base64 string. 
reference - [](https://github.com/algorand/js-algorand-sdk/blob/1d4216852ceacd0d31dad7363e31364ba3a759c7/src/encoding/address.js#L53)

Reply:
OK, please add this comments to the  code.

Reply:
Alright